### ISSUE TITLE
Question: Safe to migrate version numbers without upgrading clients

### ISSUE THREAD

**Issue by kag0** [fugue/credstash#152]
We currently have a bunch of services using credstash 1.6 to fetch values. If we run the [version migration script](https://github.com/fugue/credstash/blob/master/credstash-migrate-autoversion.py) will these old clients continue to correctly grab the latest version while we slowly update them one by one to the new client version? 

Thanks

**Comment by artburkart** [comment]
@kag0, I think the answer depends on your interpretation of "continue to correctly grab the latest version." If you mean, will credstash 1.6 continue to fetch an item with version `11` before it grabs an item with version `2`, then no, it will not continue to function as you expect. However, if you are asking if credstash 1.6 will grab `0000000000000000002` before `0000000000000000011`, then yes it will `get` items correctly by version number. 

**tl;dr:** As long as your versions remain the same when sorted lexicographically, then you'll be fine. If your versions are not integers, then no actual migration will occur. You can see more here: [README](https://github.com/fugue/credstash/blob/master/README.md#special-note-for-those-using-credstash-auto-versioning-before-december-2015).
